pyHS100/cli.py

```
@@ -80,6 +89,25 @@ def discover(ctx, timeout, discover_only):
     return found_devs
 
 
+@cli.command()
+@click.option('--timeout', default=3, required=False)
+@click.option('--attempts', default=10, required=False)
+@click.option('--devicename', required=True)
+@click.pass_context
+def find_host_from_device_name(ctx, timeout, attempts, devicename):
+    """Discover devices in the network."""
+    host = None
+    click.echo("Trying to discover %s using %s attempts of %s seconds" %
+               (devicename, attempts, timeout))
+    for attempt in range(1, attempts):
+        print("Attempt %s of %s" % (attempt, attempts))
+        found_devs = Discover.discover(timeout=timeout).items()
+        for ip, dev in found_devs:
+            if dev.alias.lower() == devicename.lower():
+                host = dev.host
+                return host
+    return None
```

Should this rather raise a corresponding exception instead of returning none?

---

I see the logic of returning none. If a device can't be found via name, the cli will continue searching for a device using discovery. On the other hand unsuccessful discovery also doesn't raise an exception, so I think we should adhere here to `discover`'s behavior.